license-maven-plugin-svn/src/main/java/com/mycila/maven/plugin/license/svn/SVNPropertiesProvider.java

```diff
@@ -107,45 +125,25 @@ public void handleLogEntry(SVNLogEntry logEntry) throws SVNException {
     };
 
     try {
-      if (!this.initialized) {
-        synchronized (this) {
-          if (!this.initialized) {
-            this.initialized = true;
-            // One-time warning for shallow repo
-            if (mojo.warnIfShallow) {
-              SVNInfo info = clientManager.getWCClient().doInfo(documentFile, SVNRevision.HEAD);
-              if (info.getDepth() != SVNDepth.INFINITY) {
-                mojo.warn("Sparse svn repository detected. Year and author property values may not be accurate.");
-              }
-            }
+      // One-time warning for shallow repo
+      if (mojo.warnIfShallow && !warnedIfShallow.get()) {
+        SVNInfo info = svnClientManager.getWCClient().doInfo(documentFile, SVNRevision.HEAD);
+        if (info.getDepth() != SVNDepth.INFINITY) {
+          if (warnedIfShallow.compareAndSet(false, true)) {
+            mojo.warn(
+                "Sparse svn repository detected. Year and author property values may not be accurate.");
```

Comment:
@dbwiddis : FYO warnedIfShallow code updated

Reply:
When I first put this in I didn't realize the SVN provider didn't have the same set of properties.  There are no authors to consider, and only the "last change" year is relevant (so a sparse repo may miss the latest commit of a file and default to current year).  So this can probably just say "year" values like the adjusted git one above.